### DISCUSSION TITLE
Qns about waldo example

### DISCUSSION THREAD

**Discussion by ongspxm** [risc0/risc0#747] [Q&A] [answered]
Hi, new to this whole zk thingy, and was trying to understand the waldo example.

If the goal is to hide the coords of the image patch, won't the merkle tree path expose the coordinates.

Given that the guest has to perform the merkleziation, they will in theory have all the required details to figure out where in the image the merkle path leads to 🤔

**Comment by flaub** [comment, marked as answer]
I think the key part here, is that the verifier (which is independent of the host or guest) won't have any knowledge of the coordinates. The guest can decide to share information by publishing it into the journal, but if it's not in the journal then it's considered private. It's true that the host and guest will have access to everything, but it's the verifier which we're trying to hide data from.

**Reply by ongspxm** [reply]
Hmmm... just to check that I got it right. So the verifier doesn't really get the actual image, rather, the verifier only has the following information.

- the root hash of the merklized image
- merkle path to the patch of image
- the patch of image
